### PR TITLE
DOC: added note for git tagging

### PR DESCRIPTION
**Pull request by agctute**
adds a note in getting started contribution documentation to prevent others from making the same mistake #23771

### CODE REVIEW

**Comment by mattip** [comment]
Shouldn't tagging releases be part of the release documentation and not part of a normal developer workflow?

**Comment by seberg** [comment]
The devdocs mention `git pull upstream main --tags` somewhere (not in the "running tests" because this has to do with building after all).  I suspect that should be enough, as Matti said, tagging yourself isn't correct.

**Comment by ngoldbaum** [comment]
It's mentioned on https://numpy.org/devdocs/dev/, but isn't mentioned in https://numpy.org/devdocs/user/building.html or on https://numpy.org/devdocs/dev/development_environment.html. The error you get is sufficiently confusing I think it's worth having this information in a few places in the docs.

**Comment by agctute** [comment]
> The devdocs mention git pull upstream main --tags somewhere (not in the "running tests" because this has to do with building after all). I suspect that should be enough, as Matti said, tagging yourself isn't correct.

I checked and you're right, the docs do mention setting upstream with tags. Unfortunately, I seem to have been using [https://numpy.org/doc/stable/dev/index.html](https://numpy.org/doc/stable/dev/index.html) instead of [numpy.org/devdocs/dev](https://numpy.org/devdocs/dev/) as my main source for trying to solve this issue, which does not mention the `--tags` flag at all

**Comment by rossbar** [comment]
> Unfortunately, I seem to have been using https://numpy.org/doc/stable/dev/index.html instead of [numpy.org/devdocs/dev](https://numpy.org/devdocs/dev/) as my main source for trying to solve this issue, which does not mention the --tags flag at all

The stable docs will be updated with this info in the next release. Prehaps we can close this then since it's already mentioned?

File: doc/source/dev/development_environment.rst [hunk 1]
```diff
@@ -64,6 +64,12 @@ This builds NumPy first, so the first time it may take a few minutes.  If
 you specify ``-n``, the tests are run against the version of NumPy (if
 any) found on current PYTHONPATH.
 
+.. note::
+
+    If the above commands return a RuntimeError with the message "Cannot parse version 0+untagged.xxxxx,"
+    it indicates that you are currently developing on a fork that lacks release tags.
+    To resolve this issue, execute the following command in your ``main`` branch: ``git pull upstream main --tags`` 
+
```
Comment (rossbar) [inline doc/source/dev/development_environment.rst:72]:
```suggestion
.. note::

    If the above commands result in ``RuntimeError: Cannot parse version...``,
    run ``git pull upstream main --tags``.

```

Just a wording suggestion to try to distill down to the critical info and conform to an 80-ish character line limit. Neither of these are blockers.

**Review by rossbar — APPROVED** [review]
Ah sorry, I had missed @ngoldbaum 's comment. This is fine with me if it helps with the development workflow!

I've left a minor wording/formatting suggestion but feel free to ignore.

File: doc/source/dev/development_environment.rst (hunk 1 above)
Comment (agctute) [reply doc/source/dev/development_environment.rst]: Thanks for checking it out! I applied most of your suggestions.

**Comment by rossbar** [comment]
Thanks @agctute !

**Pull request merged**